### DISCUSSION TITLE
Empty output from patching own font (no errors)

### DISCUSSION THREAD

**Discussion by benfrain** [ryanoasis/nerd-fonts#735] [Q&A]
I run `fontforge -script font-patcher /Users/benfrain/Downloads/ttc-iosevka-slab-11.2.2/iosevka-slab-regular.ttc -out ~/Desktop/IosevkaNerdFont/`

And I get output like this:

```
Copyright (c) 2000-2021. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Version: 20201107
 Based on sources from 2021-03-14 03:54 UTC-D.
No configfile given, skipping configfile related actions
Nerd Fonts Patcher v2.1.0 executing

Adding 56 Glyphs from Seti-UI + Custom Set
╢████████████████████████████████████████╟ 100%
Adding 198 Glyphs from Devicons Set
╢████████████████████████████████████████╟ 100%

Done with Patch Sets, generating font...

Generated: IosevkaNerdFont-Slab
```

However, I just get an empty folder with no font in it. What am I doing wrong?

**Comment by benfrain** [comment]
Wondering now if it's to do with using a ttc font? https://github.com/ryanoasis/nerd-fonts/issues/175

**Reply by Finii** [reply]
When `fontforge` opens a TrueType collection, usually you have to manually select which font you really want to open.

I never tried the script on a ttc, but what you can do is pull out the one ttf you want, save it, and patch that.

Ah, nice old issue with even a link to a ttc-patch. At the moment I'm rather busy, but that should be not hard to add. But I guess not many people would need ttc capability.

But we could of course at least bail out with a hint message :-D

**Reply by Finii** [reply]
It's in fact more complicated.

`fontforge` does open the ttc, but can not save it the way we do. But there is no diagnostics provided that is accessible from Python.

So if you do a `-ext ttf` it saves a correct font... (the one that was automatically opened / choosen from the collection).
Still weeding through fontforge source to find a way ... sigh

**Comment by benfrain** [comment]
As a workaround I ended up just using Kitty’s ability to use a different font for different ranges: https://benfrain.com/patching-another-font-for-powerline-with-kitty-you-dont-need-to/

**Reply by Finii** [reply]
Great :-)

On the other hand the symbols from the other font are not scaled to match you main font in that case.
This is just slightly different from having an unpatched font and system's font fallback in action. ;-)

**Comment by benfrain** [comment]
Oh for sure. If this can be fixed here that would be great 👍

**Comment by Finii** [comment]
#752
